quickwit/quickwit-metastore/src/metastore/file_backed/file_backed_index/shards.rs

```diff
@@ -204,8 +204,8 @@ impl Shards {
             if let Entry::Occupied(entry) = self.shards.entry(shard_id.clone()) {
                 let shard = entry.get();
                 if !force && !shard.publish_position_inclusive().is_eof() {
-                    let message = format!("shard `{shard_id}` is not deletable");
-                    return Err(MetastoreError::InvalidArgument { message });
+                    warn!("shard `{shard_id}` is not deletable");
```

Comment:
The behavior is not documented here nor in the metastore service.
This is quite important. I think we need an explanation about this corner case in the metastore.proto file.

Why did we change the behavior here? An error seemed to make sense.

I think we should not continue, log a warn and really remove the shard.

Reply:
Given the little time I have to ship this feature, I temporarily changed this behavior because it is complex to replicate for PostgreSQL.

I'm planning on following up with another PR and re-enable. I personally prefer A). You can always do B) with `force=true`